datacube_stats/utils/__init__.py

```diff
@@ -96,6 +96,21 @@ def to_float(da):
     return data.apply(to_float, keep_attrs=True)
 
 
+def restore_nodata(var_name, var, measurements):
+    assert isinstance(var, xarray.DataArray)
+
+    values = var.values
+
+    if da_is_float(var):
+        for measurement in measurements:
+            if measurement['name'] == var_name:
```

Comment:
Why not just lookup `measurements.get(var_name)` with error/warning if it's not present in the `measurements` spec.

Reply:
Not sure what you mean here. The variable `measurements` looks like:

```python
[{'name': 'BS', 'dtype': 'int8', 'nodata': -1, 'units': 'percent'}, ...]
```

so it's a `list` of `dict`s with no more structure.

Reply:
not always, sometimes it's copied from product definition, and in there `nodata` is not a mandatory field. If my dtype is float32 I shouldn't need nodata, if my data doesn't have nodata even if it's not float then I don't set nodata.